poh/benches/poh.rs

```diff
@@ -65,3 +76,129 @@ fn bench_poh_lock_time_per_batch(bencher: &mut Bencher) {
         poh.hash(DEFAULT_HASHES_PER_BATCH);
     })
 }
+
+#[bench]
+fn bench_poh_recorder_record_transaction_index(bencher: &mut Bencher) {
+    let ledger_path = get_tmp_ledger_path_auto_delete!();
+    let blockstore =
+        Blockstore::open(ledger_path.path()).expect("Expected to be able to open database ledger");
+    let GenesisConfigInfo { genesis_config, .. } = create_genesis_config(2);
+    let bank = Arc::new(Bank::new_for_tests(&genesis_config));
+    let prev_hash = bank.last_blockhash();
+
+    let (mut poh_recorder, _entry_receiver, _record_receiver) = PohRecorder::new(
+        0,
+        prev_hash,
+        bank.clone(),
+        Some((4, 4)),
+        bank.ticks_per_slot(),
+        Arc::new(blockstore),
+        &std::sync::Arc::new(LeaderScheduleCache::new_from_bank(&bank)),
+        &PohConfig::default(),
+        Arc::new(AtomicBool::default()),
+    );
+    let h1 = hash(b"hello Agave, hello Anza!");
+    poh_recorder.set_bank_with_transaction_index_for_test(bank.clone());
+    poh_recorder.tick();
+    assert_eq!(
+        poh_recorder
+            .working_bank
+            .as_ref()
+            .unwrap()
+            .transaction_index
+            .unwrap(),
+        0
+    );
+
+    bencher.iter(|| {
+        let tx0 = test_tx();
+        let tx1 = test_tx();
```

Comment:
we should avoid constructing these inside the benchmark, because this adds transaction signing into our times which is not what we want to measure here.

I'd probably just create a small buffer of transactions and clone them each iteration instead. This is realistic with how we actually record in production since we call `SanitzedTransaction::to_versioned_transaction`.

^ Actually, using that same setup is not a bad idea here. Create a buffer of `SanitizedTransaction` and then for each call to record create a vec from `txs.iter().map(|tx| tx.to_versioned_transaction()).collect()` similar to how we do in `consumer.rs` (though without the filtering)